### PR TITLE
Remove Trusty-specific logic

### LINKED ISSUE #4341: Remove Trusty-specific logic from application code and dev env
## Description

Some platform-specific logic was added to handle the upgrade from Trusty to Xenial and should be removed:
- Source Interface Submission disable logic: https://github.com/freedomofpress/securedrop/pull/4325
- Admin Interface alert for end-of-life OS: https://github.com/freedomofpress/securedrop/pull/4055

There is also some Trusty-specific logic in the development environment: 
- [Docker images](https://github.com/freedomofpress/securedrop/tree/develop/securedrop/dockerfiles/trusty)
- [CI targets](https://github.com/freedomofpress/securedrop/blob/develop/.circleci/config.yml#L59
- The use of `BASE_OS` in Makefile targets
- Upgrade testing scenarios and boxes

Since Trusty will no longer be supported after 0.12.2, we should remove this logic to improve the maintainability of the codebase and reduce the risk of errors.

### PR DESCRIPTION
**Pull request by emkll**
## Status

Ready for review

## Description of Changes

Fixes #4341 

Removes Trusty specific logic from:

* Application logic on Source and Journalist for Trusty-specific warnings
* CI targets
* Molecule Upgrade scenarios
* Molecule libvirt Scenarios
* Molecule Virtualbox scenarios
* Molecule Build scenarios

## Testing

- [ ] `make build-debs`  completes successfully (should be covered by CI)
- [ ] `molecule converge -s libvirt-staging-xenial`  completes successfully (should be covered by CI)
- [ ] `make staging` completes successfully in VirtualBox on a Mac
- [ ] `make staging`  completes successfully in Qubes
- [ ] Upgrade scenario completes successfully (`make upgrade-start` and  `make upgrade-test-local`)
- [ ] There are no more references to Trusty that need to be removed

## Deployment

1. Changes to Source and Journalist interfaces will be shipped via the securedrop-app-code package
2. All other changes are dev/ci only

### If you made non-trivial code changes:

- [x] I have written a test plan and validated it for this PR

### CODE REVIEW

File: securedrop/source_app/decorators.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ def ignore_static(f):
     a static resource."""
     @wraps(f)
     def decorated_function(*args, **kwargs):
-        if request.path.startswith('/static') or request.path == '/org-logo':
```
Comment (redshiftzero) [inline securedrop/source_app/decorators.py]: why remove this `/org-logo` logic?

Comment (ninavizz) [reply securedrop/source_app/decorators.py]: Actually... removing that logic throughout the Source UI, has been an ongoing desire of mine for some time. 

There's no intuitive destination imho for that logo to point to; and that functionality was put in before "Cancel" buttons existed on all the pages as 'the way out.' Likewise, its `<alt>` tag should be changed to simply say "Logo" and not "SecureDrop." (...as the latter is _very_ confusing, especially when it's all that displays as Tor slowly loads the org logo)

Comment (redshiftzero) [reply securedrop/source_app/decorators.py]: sorry nina this is just logic regarding how URLs are routed through the application, not actually removing the logo logic in its entirety 😇 

but OK I see why this is being done in 95c0c31fdc38eafc60932d7eec1e6ab57c290025, resolving this convo

Comment (ninavizz) [reply securedrop/source_app/decorators.py]: *blushes* roger that! 😸

**Comment by redshiftzero** [comment]
after the change in 4376e0f, we'll also need to update the references in `virtualbox-staging-xenial`:

```diff
diff --git a/molecule/virtualbox-staging-xenial/molecule.yml b/molecule/virtualbox-staging-xenial/molecule.yml
index 9ba779328..51dfe2322 100644
--- a/molecule/virtualbox-staging-xenial/molecule.yml
+++ b/molecule/virtualbox-staging-xenial/molecule.yml
@@ -41,9 +41,9 @@ provisioner:
     e: "@ansible-override-vars.yml"
   playbooks:
     converge: ../../install_files/ansible-base/securedrop-staging.yml
-    create: ../libvirt-staging/create.yml
-    destroy: ../libvirt-staging/destroy.yml
-    prepare: ../libvirt-staging/prepare.yml
+    create: ../libvirt-staging-xenial/create.yml
+    destroy: ../libvirt-staging-xenial/destroy.yml
+    prepare: ../libvirt-staging-xenial/prepare.yml
   env:
     ANSIBLE_CONFIG: ../../install_files/ansible-base/ansible.cfg
```

**Comment by rmol** [comment]
I've completed these tests:

- [x] make build-debs completes successfully (should be covered by CI)
- [x] molecule converge -s libvirt-staging-xenial completes successfully (should be covered by CI)
- [x] Upgrade scenario completes successfully (make upgrade-start and make upgrade-test-local)

**Comment by redshiftzero** [comment]
Mac specific testing:

- [x] `make build-debs`  completes successfully on mac (modulo known issues)
- [x] `make staging` completes successfully in VirtualBox on a Mac

**Comment by emkll** [comment]
a5e4718dfd95311936f92928eb192182b9ecd4be looks good to me :+1:

**Comment by zenmonkeykstop** [comment]
just finished running `make staging` on Qubes - a bit 12th-hour but I can confirm that it works :)

**Pull request merged**